semantic-python/src/Language/Python/Core.hs

```diff
@@ -194,7 +194,21 @@ instance Compile Py.Call where
     locate it (func $$* args) & cc
   compile it _ _ = fail ("can't compile Call node with generator expression: " <> show it)
 
-instance Compile Py.ClassDefinition
+instance Compile Py.ClassDefinition where
+  compile it@Py.ClassDefinition { body = pybody, name = Py.Identifier _ann (Name -> n) } cc next = do
+    let buildTypeCall _ = do
+          bindings <- asks @Bindings (toList . unBindings)
+          let buildName n = (n, pure n)
+              contents = record . fmap buildName $ bindings
+              typefn = pure "__semantic_prelude" ... "type"
+
+          pure (typefn $$ Core.string (coerce n) $$ pure "object" $$ contents)
```

Comment:
_Python: The Full Monty_ points out that
> Most Python programmers use the special `class` form to create classes in Python. However, `class` is merely syntactic sugar for a use of the builtin Python function `type`. The [documentation](https://docs.python.org/3/library/functions.html#type) states explicitly that the following forms are equal:

```python
class X: a=1

X = type(’X’, (object,), dict(a=1))
```

This means that the body of classes - essentially everything interesting about them - is capable of being passed in a record, as the final parameter. This is very close to the semantics of top-level module definitions, which also keep track of internal definitions and ultimately return a record. As such, we can emulate our implementation: rather than passing in `pure` as the continuation for the compilation of the class body, we can pass in a function that `asks` for the set of recorded bindings and then stuffs them all in a record passed as a parameter to `type` (as defined in the Prelude, since it’s legal for Python programs to assign to `type` and some testing shows that class definitions still work if you set `type = None`).

The utility of continuations continues to impress me.